client/src/App.tsx

```diff
@@ -24,9 +24,9 @@ const theme = createMuiTheme({
 })
 
 const bearerToken = {
-  tokenFromQueryParamsRegexp: /token=([^&]+)/,
+  tokenFromQueryParamsRegexp: /token=(.*)/,
   extractToken(): string | undefined {
-    const match = document.location.search.match(/token=(\w+)/)
+    const match = document.location.search.match(/token=(.*)/)
```

Comment:
/facepalm

That regex works, I just forgot to use it. LOL.

So, that code was supposed to be:
```ts
    const match = document.location.search.match(this.tokenFromQueryParamsRegexp)
```

**Why extract the regexp to a variable instead of just writing it directly in the `match()`?**

Cause the expensive part about regex is creating them, not running them.

**Why not `/token=(.*)/`?**

Using `.` instead of `[^&]` matches absolutely anything, so if we somehow had another query parameter in the same URL, we would capture too much. For example if we had `?token=some.token.here&anotherParam=123`. 

Using `*` instead of `+` matches even when there's 0 instances, but we should not match if the token is not present. Example: `?token=` should not be a match.